pkg/cli/config/config.go

```diff
@@ -351,3 +352,56 @@ func getRawKubeConfigLoader(kubeconfigPath string) clientcmd.ClientConfig {
 	// Return the clientcmd.ClientConfig (equivalent to ToRawKubeConfigLoader)
 	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
 }
+
+// ServerConnectionFlagsChanged returns true if any server connection flags are set.
+func ServerConnectionFlagsChanged(cmd *cobra.Command) bool {
+	return cmd.Flags().Changed("host") ||
+		cmd.Flags().Changed("token") ||
+		cmd.Flags().Changed("insecure-skip-tls-verify") ||
+		cmd.Flags().Changed("api-path")
+}
+
+// BuildDirectClientConfig builds a client.Config from CLI flags (host, token, api-path, insecure-skip-tls-verify).
+func BuildDirectClientConfig(p common.Params) (*client.Config, error) {
+	host := p.Host()
+	token := p.Token()
+	if host == "" || token == "" {
+		return nil, errors.New("--host and --token flag must be set if using direct connection flags")
+	}
+	rc := &rest.Config{
+		Host: host,
+	}
+	if p.APIPath() != "" {
+		rc.APIPath = p.APIPath()
+	}
+	if p.Token() != "" {
+		rc.BearerToken = token
+	}
+
+	rc.Insecure = p.SkipTLSVerify()
+	// Optionally set timeout (default 30s)
+	rc.Timeout = 30 * time.Second
```

Comment:
I consider that default value low. I've witnessed some busy instances where it takes more than 30 second to get a response from the Results API service. Maybe 60 second. Even better if the user can modify that from the command line, could be done in a separate PR.